src/topology/continuous_function/bounded.lean

```diff
@@ -1030,4 +1032,55 @@ module over the algebra of bounded continuous functions from `α` to `𝕜`. -/
 
 end normed_algebra
 
+section cstar_ring
+/-!
+### Star structure
+
+In this section, if `β` is a C⋆-ring, then the space of bounded
+continuous functions from `α` to `β` inherits a C⋆-ring structure also,
+by using the star operation pointwise. Moreover, if `𝕜` is normed field
+and a ⋆-ring, over which `β` is a normed algebra and a star module, then
+the space of bounded continuous functions from `α` to `β` is a star module
+over `𝕜`. In other words, `α →ᵇ β` is a C⋆-algebra (completeness is
+guaranteed when `β` is complete). -/
+
+variables {𝕜 : Type*} [normed_field 𝕜] [star_ring 𝕜]
+variables [topological_space α]
+variables [normed_ring β] [star_ring β] [normed_star_monoid β]
+variables [normed_algebra 𝕜 β] [star_module 𝕜 β]
+
+instance : star_ring (α →ᵇ β) :=
+{ star := λ f, f.comp star star_normed_group_hom.lipschitz,
+  star_involutive := λ f, ext $ λ x, star_star (f x),
+  star_mul := λ f g, ext $ λ x, star_mul (f x) (g x),
+  star_add := λ f g, ext $ λ x, star_add (f x) (g x) }
```

Comment:
Can you separate out the `star_monoid` instance that doesn't need `normed_ring β`, as:
```lean
instance [topological_space α] [normed_group β] [star_add_monoid β] [normed_star_monoid β] :
  star_add_monoid (α →ᵇ β) :=
{ star := λ f, f.comp star star_normed_group_hom.lipschitz,
  star_involutive := λ f, ext $ λ x, star_star (f x),
  star_add := λ f g, ext $ λ x, star_add (f x) (g x) }
```

Reply:
The `star_module` then also only needs `normed_group β`

Reply:
Done. I added a `normed_star_monoid` instance as well.